relayer-cli/src/commands/query/channel_ends.rs

```diff
@@ -0,0 +1,182 @@
+    channel_end: ChannelEnd,
+    connection_end: ConnectionEnd,
+    client_state: ClientState,
+    counterparty_channel_end: ChannelEnd,
+    counterparty_connection_end: ConnectionEnd,
+    counterparty_client_state: ClientState,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct TraceSummary {
+    chain_id: ChainId,
+    client_id: ClientId,
+    connection_id: ConnectionId,
+    channel_id: ChannelId,
+    port_id: PortId,
+
+    counterparty_chain_id: ChainId,
+    counterparty_client_id: ClientId,
+    counterparty_connection_id: ConnectionId,
+    counterparty_channel_id: ChannelId,
+    counterparty_port_id: PortId,
+}
+
+fn do_run(cmd: &QueryChannelEndsCmd) -> Result<(), Box<dyn std::error::Error>> {
+    debug!("Options: {:?}", cmd);
+
+    let config = app_config();
+
+    let chain_id = cmd.chain_id.clone();
+    let port_id = cmd.port_id.clone();
+    let channel_id = cmd.channel_id.clone();
+
+    let chain_config = config
+        .find_chain(&chain_id)
+        .ok_or_else(|| format!("chain '{}' not found in configuration file", chain_id))?;
+
+    let rt = Arc::new(TokioRuntime::new().unwrap());
+    let chain = CosmosSdkChain::bootstrap(chain_config.clone(), rt.clone())?;
+
+    let chain_height = match cmd.height {
+        Some(height) => Height::new(chain.id().version(), height),
+        None => chain.query_latest_height()?,
+    };
+
+    let channel_end = chain.query_channel(&port_id, &channel_id, chain_height)?;
+
+    let connection_id = channel_end
+        .connection_hops
+        .first()
+        .ok_or_else(|| format!("missing connection_hops"))?
+        .clone();
+
+    let connection_end = chain.query_connection(&connection_id, chain_height)?;
+
+    let client_id = connection_end.client_id().clone();
+
+    let client_state = chain.query_client_state(&client_id, chain_height)?;
+
+    let channel_counterparty = channel_end.counterparty().clone();
+    let connection_counterparty = connection_end.counterparty().clone();
+
+    let counterparty_client_id = connection_counterparty.client_id().clone();
+
+    let counterparty_connection_id = connection_counterparty.connection_id.ok_or_else(|| {
+        format!(
+            "connection end do not have counterparty connection id: {:?}",
+            connection_end
+        )
+    })?;
+
+    let counterparty_port_id = channel_counterparty.port_id().clone();
+
+    let counterparty_channel_id = channel_counterparty.channel_id.ok_or_else(|| {
+        format!(
+            "channel end do not have counterparty channel id: {:?}",
+            channel_end
+        )
+    })?;
+
+    let counterparty_chain_id = client_state.chain_id.clone();
+
+    let chain_config_b = config.find_chain(&counterparty_chain_id).ok_or_else(|| {
+        format!(
+            "chain '{}' not found in configuration file",
+            counterparty_chain_id
+        )
+    })?;
+
+    let counterparty_chain = CosmosSdkChain::bootstrap(chain_config_b.clone(), rt).unwrap();
```

Comment:
It would be ideal to avoid `unwrap`. Perhaps try using `spawn_chain_runtime(..)?`.

Reply:
Fixed.